packages/compiler/src/compiler.ts

```diff
@@ -246,7 +256,15 @@ export class ComponentCompiler {
     // wait on CSS initialization
     await this.cssParser.init();
 
-    const componentPath = componentId.split('##')[0];
+    let [componentPath] = componentId.split('##');
```

Comment:
By checking here, we're only warning for the root Component. Let's add it to `this.getComponentSources` instead to catch imports at the source.